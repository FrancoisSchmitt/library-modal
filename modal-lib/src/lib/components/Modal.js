import React from "react";
import "./modal.css";

const Modal = ({
    title ,
    message,
    btnMessage,
    openModal,
    closeModal,

}) =>
(
    <div className="modal-container">
        {
            openModal === true &&
            <div className='modal'>
                <div className="modal-content">
                    <div className='modal-title-content'>
                        <h3 className="modal-title">{title}</h3>
                        <p className='modal-paragraphe'>{message}</p>
                        <button onClick={closeModal} className="btn-close">{btnMessage}</button>
                    </div>
                </div>
            </div>

            }
     </div>
);

export default Modal;