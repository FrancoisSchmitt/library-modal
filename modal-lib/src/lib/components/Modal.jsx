import React from "react";
import "./modal.css";

const TextInput = ({
    title ,
    message,
    btnMessage,
    openModal,
    closeModal,

}) => (
    <div>
        <h1>lol
            
        </h1>
        {
            openModal === true &&
            <div className='Modal'>
                <div className="modal-section"></div>
                <div className="modal-content">
                    <div className='modal-title-content'>
                        <div className='modal-title'>
                            <h3>{title}</h3>
                            <button onClick={closeModal} className="btn-close-icon"><i className="fa fa-times"></i></button>
                        </div>
                        <p className='modal-paragraphe'>{message}</p>
                        <button onClick={closeModal} className="btn-close">{btnMessage}</button>
                    </div>
                </div>
            </div>
        }
    </div>
);

export default TextInput;