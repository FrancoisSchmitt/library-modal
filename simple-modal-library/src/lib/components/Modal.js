import './index.css'
import 'font-awesome/css/font-awesome.min.css';
export default function Modal({ openModal, closeModal, message, title, btnMessage }) {
    // Template
openModal = true
    return (
        <>
            {
    openModal === true &&
        <div className='Modal'>
            <div className="modal-section"></div>
                <div className="modal-content">
                    <div className='modal-title-content'>
                        <div className='modal-title'>
                            <h3>{title}</h3>
                            <button onClick={closeModal} className="btn-close-icon"><i className="fa fa-times"></i></button>
                        </div>
                        <p className='modal-paragraphe'>{message}</p>
                                <button onClick={closeModal} className="btn-close">{btnMessage}</button>
                </div>
            </div>
        </div> 
            }
        </>
    );
}

