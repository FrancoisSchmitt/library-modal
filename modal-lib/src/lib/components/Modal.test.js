import React from "react";
import Modal from "./Modal";
import renderer from "react-test-renderer";

describe("Modal", () => {
  it("renders properly", () => {
    const tree = renderer
      .create(<Modal title="Your title modal" Message="Your message from the modal" btnMessage="Submit"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
