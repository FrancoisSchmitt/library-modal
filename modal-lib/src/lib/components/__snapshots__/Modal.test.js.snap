// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders properly 1`] = `
<div
  className="simple-form-group"
>
 <>
        {
            openModal === true &&
            <div className='Modal'>
                <div className="modal-section"></div>
                <div className="modal-content">
                    <div className='modal-title-content'>
                        <div className='modal-title'>
                            <h3 title="simple-text-input"></h3>
                            <button onClick={[function]} className="btn-close-icon"><i className="fa fa-times"></i></button>
                        </div>
                        <p className='modal-paragraphe' message="simple-text-input"></p>
                        <button onClick={[function]} className="btn-close" btnMessage="simple-text-input"></button>
                    </div>
                </div>
            </div>
        }
    </>

`;
